pages/cookbook/jettons.mdx

```diff
@@ -1,9 +1,161 @@
 # Fungible Tokens (Jettons)
 
-import { Callout } from 'nextra/components'
+import { Callout } from 'nextra-theme-docs';
 
-<Callout emoji="🚨">
+This page lists common examples of working with [jettons](https://docs.ton.org/develop/dapps/asset-processing/jettons).
 
-  This page is a stub. [Contributions are welcome!](https://github.com/tact-lang/tact-docs/issues)
+## Accepting jetton transfer
+
+Transfer notification message have the following structure.
+
+```tact
+message(0x7362d09c) TransferNotification {
+    queryId: Int as uint64;
+    jettonAmount: Int as coins;
+    fromAddress: Address;
+    forwardPayload: Slice as remaining;
+}
+```
+
+Use [receiver](/book/receive) function to accept token notification message.
+
+<Callout type="warning" emoji="⚠️">
+
+  Sender of transfer notification must be validated!
+
+</Callout>
+
+Validation can be done using jetton wallet state init and calculating jetton address.
+Note, that notifications are coming from YOUR contract's jetton wallet, so [`myAddress()`](/ref/core-common#myaddress) should be used in owner address field.
+Wallet initial data layout is shown below, but sometimes it can differ.
```

Comment:
we should also mention that in many common cases it's enough to add some contract storage field like `myJettonWallet` and calculate the address offchain upon deploying the contract. Calculating it on-chain every time is not that good in terms of gas usage and it'll also require customizations for jettons with modified code.

it's already mentioned for sending jettons below, but it should also be mentioned here.

Reply:
Resolved